scripts/git/check_submodule_init.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import os
+import subprocess
+import sys
+
+
+def run():
+  # No-op if we're not in a git repository.
+  try:
+    subprocess.check_call(['git', 'rev-parse', '--is-inside-work-tree'],
+                          stdout=subprocess.DEVNULL,
+                          stderr=subprocess.DEVNULL)
+  except:
+    return
+
+  output = os.popen("git submodule status")

```

Comment:
Oh, and since this can take ~2 seconds to run, I'm considering caching the result and only checking once the git commit changes. That would be extra code that we can work on after this is merged.

https://cmake.org/cmake/help/latest/module/FindGit.html
https://cmake.org/pipermail/cmake/2018-October/068388.html
https://cmake.org/cmake/help/latest/command/set.html
https://cmake.org/cmake/help/latest/command/unset.html

pseudocode (cmake):
```
if in git repo:
  get short git commit hash
  if hash =/= cached hash
    run script
    updated cached hash
```

Reply:
Yes @ScottTodd , `git submodule status` is taking some time. We can use caching and I will be glad if I get a chance to contribute again. Thanks for merging the pr.